### DISCUSSION TITLE
Busted launch today

### DISCUSSION THREAD

**Discussion by apoc527** [ttscoff/bunch-docs#8] [General]
Since Brett asked for more discussions, I thought I'd start one. Of course, I just got the "update now" message, so this could all be moot. But this morning, I rebooted and Bunch Beta got stuck. I *think* it was related to the "Ask" popup for one of my bunches but there was no indication that was the case until I tried just clicking on the Bunch dock icon, finally got the "Load XX" and hit yes, THEN it completed its launch sequence.

**Comment by ttscoff** [comment]
Yeah, what happens is that the dialog gets hidden behind other windows. Because it's modal, everything stops until you answer. So if you can't see and dismiss the dialog, then the application is basically frozen. I need to see if I can force those alerts above all other windows and keep them there. I don't think NSAlerts behave like windows and allow setting level like that, but I'm working on a solution.

This is especially bad if you're running in menu bar mode, because the app can't become foreground and raise its dialogs. With Dock mode you can click the icon and find them, but in menu bar mode they're lost until you move all your other windows and reveal them.

**Comment by ttscoff** [comment]
I opened ticket #13. Then closed it because I think I figured out how to make sure the dialogs don't get lost. I'll get the update with this and the menu ordering fix out today, so let me know if b68 doesn't fix it after you update.

**Comment by ttscoff** [comment]
Hey @apoc527 is this fixed in more recent builds for you?

**Reply by apoc527** [reply]
Yes, mostly, insofar as the app window is more reliably out in front. There's a new issue with Fantastical where it fails to launch. So far it's pretty consistently just Fantastical that won't launch. I don't think it's because the Fantastical menu bar helper is already running, but maybe?

**Reply by ttscoff** [reply]
Testing on Big Sur with the latest Fantastical it launches and quits fine for me with the menu bar item running. The main window comes up and closes with the Bunch.

**Reply by apoc527** [reply]
It failed because i'm a doofus who didn't notice the typo. 
<img width="237" alt="CleanShot 2021-03-26 at 09 29 31@2x" src="https://user-images.githubusercontent.com/24831696/112663229-cf66a480-8e15-11eb-8294-96a246798721.png">

😳

**Reply by ttscoff** [reply]
🤣
